nservicebus/sqlserver/multiple-databases.md

```diff
@@ -14,24 +14,50 @@ The SQL Server transport allows you to select, on per-endpoint basis, where the
+NOTE: If the destination endpoint uses different database or server instance, sending a message to it might cause the transaction to escalate to a distributed transaction. Usually it is not a desired effect. Use NServiceBus [Outbox](/nservicebus/outbox/) to avoid it.
 
 ## Single database
 
-By default SQL Server transport uses a single instance of the SQL Server to maintain all the queues for all endpoints of a system. In order to send a message, an endpoint needs to connect to the (usually remote) database server and execute a SQL command. The message is delivered directly to the destination queue without any store-and-forward mechanism. Such a simplistic approach can only be only used for small-to-medium size systems because of the need to store everything in a single database. Using schemas to distingiush logical data stores might be a good compromise for mid-size projects. The upside is it does not require Distributed Transaction Coordinator (MS DTC). Another advantage is the ability to take a snapshot of entire system state (all the queues) by backing up a database. This is even more useful if the business data are stored in the same database.
+By default the SQL Server transport uses a single instance of the SQL Server to maintain queues for all endpoints in the system. In order to send a message, an endpoint needs to connect to the (usually remote) database server and execute a SQL command. The message is delivered directly to the destination queue without any store-and-forward mechanism. 
 
+Using a single database doesn't require Distributed Transaction Coordinator (MS DTC). Another advantage is the ability to take a snapshot of entire system state (all the queues) by backing up a database. This is most useful when the business data is also stored in the same database.
 
-## Database-per-endpoint
+## Single database with multiple schemas
 
-In a more complex scenario endpoints use separate databases and DTC is involved. Due to lack of store-and-forward mechanism, if a remote endpoint's database or DTC infrastructure is down, the endpoint cannot send messages to it. This potentially renders the endpoint (and all endpoints transitively depending on it) unavailable.
+The default schema used by SQL Server transport is `dbo`. A different schema might be specified using API:
 
+snippet:sqlserver-singledb-multischema
 
-## Database-per-endpoint with store-and-forward
+or using a configuration file:
+
+snippet:sqlserver-singledb-multischema-config
+
+If two endpoints use different schemas then additional configuration is required. The sender needs to know the schema of the receiver, and subscriber needs the schema of the publisher. 
+
+The schema for another endpoint can be specified at compile time:
```

Comment:
I left it there, it says exactly this "Push mode: In the push mode the whole collection of endpoint connection information objects is passed during configuration time"

I'm not sure about this, Push vs Pull means nothing to the user without context. If we want to change that then I think we should explain earlier what we mean by Push and Pull and I didn't see a good way to do it in that context. It makes sense when we have two options of code configuration, but I find confusing calling config file option pull and code option push.